src/test/scala/com/normation/rudder/datasources/UpdateHttpDatasetTest.scala

```diff
@@ -253,279 +268,351 @@ class UpdateHttpDatasetTest extends Specification with BoxSpecMatcher with Logga
     )
   // create a copy of template, updating some properties
   def NewDataSource(
-      name   : String
-    , url    : String = httpDatasourceTemplate.url
-    , path   : String = httpDatasourceTemplate.path
-    , schedule: DataSourceSchedule = datasourceTemplate.runParam.schedule
-    , method: HttpMethod = httpDatasourceTemplate.httpMethod
-    , params: Map[String, String] = httpDatasourceTemplate.params
-    , headers: Map[String, String] = httpDatasourceTemplate.headers
+      name     : String
+    , url      : String              = httpDatasourceTemplate.url
+    , path     : String              = httpDatasourceTemplate.path
+    , schedule : DataSourceSchedule  = datasourceTemplate.runParam.schedule
+    , method   : HttpMethod          = httpDatasourceTemplate.httpMethod
+    , params   : Map[String, String] = httpDatasourceTemplate.params
+    , headers  : Map[String, String] = httpDatasourceTemplate.headers
+    , onMissing: MissingNodeBehavior = httpDatasourceTemplate.missingNodeBehavior
   ) = {
-    val http = httpDatasourceTemplate.copy(url = url, path = path, httpMethod = method, params = params, headers = headers)
+    val http = httpDatasourceTemplate.copy(url = url, path = path, httpMethod = method, params = params, headers = headers, missingNodeBehavior = onMissing)
     val run  = datasourceTemplate.runParam.copy(schedule = schedule)
-    datasourceTemplate.copy(name = DataSourceName(name), sourceType = http, runParam = run)
+    datasourceTemplate.copy(id = DataSourceId(name), sourceType = http, runParam = run)
 
   }
 
   object MyDatasource {
-
     val infos = new TestNodeRepoInfo(NodeConfigData.allNodesInfo)
     val http = new HttpQueryDataSourceService(
         infos
       , parameterRepo
       , infos
       , interpolation
     )
-
-
     val uuidGen = new StringUuidGeneratorImpl()
-
-
   }
 
 
   sequential
 
-  "Update on datasource" should {
-    val datasource = NewDataSource(
-        name = "test-scheduler"
-      , url  = s"${REST_SERVER_URL}/$${rudder.node.id}"
-      , path = "$.hostname"
-      , schedule = Scheduled(5.minute)
-    )
-    val action = (c: UpdateCause) => {
-      // here we need to give him the default scheduler, not the test one,
-      // to actually have the fetch logic done
-      MyDatasource.http.queryAll(datasource, c) match {
-        case Full(res) => //ok
-        case x         => logger.error(s"oh no! Got a $x")
+//  "Update on datasource" should {
```

Comment:
nope :) it was because it took to long when testing, and eclipse does not help a lot to start only one test